clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -1090,6 +1090,23 @@ public interface UncheckedCloseable extends AutoCloseable {
         void close();
     }
 
+    /**
+     * Closes {@code maybeCloseable} if it implements the {@link AutoCloseable} interface,
+     * and if an exception is thrown, it is logged at the WARN level.
+     * <b>Be cautious when passing method references as an argument.</b> For example:
+     * <p>
+     * {@code closeQuietly(task::stop, "source task");}
+     * <p>
+     * Although this method gracefully handles null {@link AutoCloseable} objects, attempts to take a method
+     * reference from a null object will result in a {@link NullPointerException}. In the example code above,
+     * it would be the caller's responsibility to ensure that {@code task} was non-null before attempting to
+     * use a method reference from it.
+     */
+    public static void maybeCloseQuietly(Object maybeCloseable, String name) {
```

Comment:
This is overkill for a single use (which is the case in this PR), but a second use case is added for it in https://github.com/apache/kafka/pull/14309.

Reply:
I found more use-cases for this today, so 👍 on adding this helper.